### DISCUSSION TITLE
Unable to connect to docker engine on lima when restarting Mac and lima

### DISCUSSION THREAD

**Discussion by ringomax** [lima-vm/lima#762] [Q&A] [answered]
Unable to connect to docker engine on lima when restarting Mac and lima
-----
### details of the error.
> `error during connect: Get "http://%2FUsers%2Fam%2F.lima%2Fdocker%2Fsock%2Fdocker.sock/v1.24/images/json": EOF`

### Version
> - lima 0.9.1
> - docker 20.10.13
> - Docker Compose 2.2.3
> - MacOS Monterey 12.3

### How the error occurs
> 1. Start lima using [Docker.yml](https://github.com/akihiro2526/docker-with-lima)
> 2.  Launching containers with Docker compose
> 3. Restart Mac with container and lima running
> 4. Restart the Mac and start lima with the command `limactl start docker`
> 5. When type the docker command, get an error code.
> `error during connect: Get "http://%2FUsers%2Fam%2F.lima%2Fdocker%2Fsock%2Fdocker.sock/v1.24/images/json": EOF`

**Status of the Docker engine on lima at the time of the error**
```
$ service docker status
○ docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
TriggeredBy: ● docker.socket
       Docs: https://docs.docker.com
```
**Restart Docker engine on lima**
```
$ sudo service docker restart
$ service docker status
● docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; disabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-03-27 02:48:57 UTC; 2s ago
TriggeredBy: ● docker.socket
       Docs: https://docs.docker.com
   Main PID: 2328 (dockerd)
      Tasks: 8
     Memory: 26.5M
        CPU: 188ms
     CGroup: /system.slice/docker.service
             └─2328 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
```
After restarting the Docker engine on lima, I get the same error and cannot connect to the Docker engine.

How can I make sure that the connection to the Docker engine does not get disconnected when I restart my Mac?

**Comment by afbjorklund** [comment]
Normally lima starts the user docker (not the system docker), so you would have to use `systemctl --user docker status`.

```console
anders@lima-docker:/home/anders/lima$ systemctl --system status docker
○ docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
TriggeredBy: ● docker.socket
       Docs: https://docs.docker.com
anders@lima-docker:/home/anders/lima$ systemctl --user status docker
● docker.service - Docker Application Container Engine (Rootless)
     Loaded: loaded (/home/anders.linux/.config/systemd/user/docker.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-03-27 08:26:01 UTC; 25min ago
       Docs: https://docs.docker.com/go/rootless/
   Main PID: 1410 (rootlesskit)
...
```

----

And it is $XDG_RUNTIME_DIR/docker.sock that is forwarded:

```yaml
portForwards:
- guestSocket: "/run/user/{{.UID}}/docker.sock"
  hostSocket: "{{.Dir}}/sock/docker.sock"
```

See https://docs.docker.com/engine/security/rootless/ for details.

It is supposed to be "enabled" (start at boot) by default, though?

**Comment by afbjorklund** [comment, marked as answer]
Separating between the user containers (a.k.a. rootless, fakeroot) and the system containers would be a good entry for the lima manual.

It is something that is often confusing to new users, especially when Docker didn't use `sudo` but instead added the `docker` root group...

**Comment by ringomax** [comment]
Thank you for your response.
